tests/lib/send_results.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+
+# This script will store the results into csv files in the
+# metrics 'results' directory.
+# The csv file names are derived from the reported 'test name', and all
+# results for a single 'test name' are collated (appended) into a single csv.
+
+# General env
+MYSELF="${0##*/}"
+SCRIPT_PATH=$(dirname $(readlink -f $0))
+RESULT_DIR="${SCRIPT_PATH}/../metrics/results"
+GROUP="PNP"
+
+declare TEST_NAME
+declare RESULT
+declare ARGS
+declare UNITS
+declare SYS_VERSION
+declare SYSTEM
+declare TAG
+declare HW
+declare SEND
+declare KERNEL
+declare IMG
+
+# Parse the platform name
+function get_platform_name()
+{
+	com="core"
+
+	model="$(cat /proc/cpuinfo | grep "model name" | uniq | \
+		awk -F ": " '{print $2}' | sed s/\(R\)//g | \
+		sed s/\(TM\)//g | sed s/[[:space:]]*CPU[[:space:]]*/" "/g | \
+		cut -d"@" -f1 | sed s/[[:space:]]*$//g)"
+
+	num_cores=$(nproc)
+
+	if (( $num_cores > 1 ));then
+		com="cores"
+	fi
+
+	echo "$model ($num_cores $com)"
+}
+
+function save_to_csv()
+{
+	runtime="/usr/bin/cc-oci-runtime"
+	release_file="/usr/lib/os-release"
+	timestamp="$(date +%s)"
+	cc_img_path="/usr/share/clear-containers"
+
+	if [ -z "$TEST_NAME" ];then
+		echo "error: test name no found"
+		exit 1;
+	fi
+
+	if [ -z "$RESULT" ];then
+		echo "error: result no found"
+		exit 1;
+	fi
+
+	if [ -z "$UNITS" ];then
+		echo "error: units no found"
+		exit 1;
+	fi
+
+	if [ -z "$HW" ];then
+		platform="$(get_platform_name)"
+		HW="$platform"
+	fi
+
+	if [ -z "$IMG" ];then
+		IMG="$(readlink "$cc_img_path/clear-containers.img")"
+	fi
+
+	if [ -z "$KERNEL" ];then
+		KERNEL="$(readlink "$cc_img_path/vmlinux.container")"
+	fi
+
+	if [ -z "$TAG" ];then
+		if [ -f "$runtime" ];then
+			commit="$(cc-oci-runtime -v | tail -n1 | sed s/"commit: "//g)"
+			TAG="$commit"
+		fi
+	fi
+
+	if [ -z "$SYSTEM" ];then
+		os="$(cat "$release_file" | grep -w "ID" | cut -d"=" -f2)"
+		SYSTEM="$os"
+	fi
+
+	if [ -z "$SYS_VERSION" ];then
+		version="$(cat "$release_file" | grep "VERSION_ID"| cut -d "=" -f2 | sed s/\"//g)"
+		if [ -z "$version" ];then
+			version="0"
+		fi
+		SYS_VERSION=$version
+	fi
+
+	if [ -z "$ARGS" ];then
+		ARGS="none"
+	fi
+
+	# Generate the file name from the test name - replace spaces to hyphens
+	CSV_FILE=${RESULT_DIR}/$(echo ${TEST_NAME} | sed 's/ /-/g').csv
+
+	# Are we dry-running?
+	if [ -z "$SEND" ];then
```

Comment:
I like the idea of having a dry-run mode, but I'm not really clear what it buys us here. I think it would just perform the checks? It might be worth having it "almost" run the code below by showing `would run '...'` using an `echo $cmd` and if not in dry-run mode, running `eval $cmd` instead.